sample_app/app/controllers/users_controller.rb

```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user,   only: [:edit, :update]
```

Comment:
i don't love how this is named.  I get the idea (user can only edit themselves), but it's not totally clear from the name.

This is also a downside of Rails' `before_action`s... it separates this security concern from the meat of the `edit` and `update` routes